app/components/GenericOauthPlatform.tsx

```diff
@@ -155,11 +155,11 @@ export const GenericOauthPlatform = ({
   async function initiateFetchCredential() {
```

Comment:
What if we move the logic for getting the proof to the providers?
We have already build exception for Twitter and Facebook (and I would hate to add more in the future). I think this component should be as simple as possible and not include logic designed for specific platforms.
I can imagine for example that we request the `accessTokenRequest` for every provider, and the providers can then implement this as required by that specific platform.
And we will only call:
1. accessTokenRequest
2. fetchCredentials
like for the facebook platform.

Reply:
I like that idea :) I can start getting that going. So it would basically be a getProof function that exists on the platform that would be called from the front end?